modules/common/added/scripts/launch.sh

```diff
@@ -40,9 +40,12 @@ if [ -z ${SPARK_MASTER_ADDRESS+_} ]; then
     exec $SPARK_HOME/bin/spark-class$JAVA_AGENT org.apache.spark.deploy.master.Master
 else
     echo "Starting worker$metrics, will connect to: $SPARK_MASTER_ADDRESS"
+
+    # spark://x.y.z:7077  -> x.y.z/7077
+    _MASTER_HOST_AND_PORT=$(echo $SPARK_MASTER_ADDRESS | sed -r 's;.*//(.*):(.*);\1/\2;g')
     while true; do
         echo "Waiting for spark master to be available ..."
-        curl --connect-timeout 1 -s -X GET $SPARK_MASTER_UI_ADDRESS > /dev/null
+        timeout 1 sh -c "(</dev/tcp/$_MASTER_HOST_AND_PORT) &>/dev/null"
```

Comment:
i'm not familiar with how you are using the device node here. could you explain it a little, or is there a doc i could read about this.

Reply:
`</dev/tcp/host/port`

..it's a posix way to find out if something is listening on that port. Netcat or telnet are probably better, but they are not present on that image and it would make the image even bigger. For example try `</dev/tcp/localhost/631` - cups server vs `</dev/tcp/localhost/632` probably nothing.

`timeout 1 sh -c "..."`
..wait only 1 second for the command, otherwise return non-zero code (`timeout 1 sh -c "sleep 5"` for instance returns `124`)

`&>/dev/null`
..shhh :)

Reply:
thanks for the breakdown, i hadn't seen the `/dev/tcp/...` usage before